src/js/browser-plugin.js

```diff
@@ -1,390 +1,546 @@
 
-      // If the result status has not reached the 'done' status then make a second API call to retrieve the 
-      // result with a maximum of 10 retries
-      if(firstCallStatus.localeCompare('done') !== 0) {
+      tweetInfo.domObject.coInfoCounter = 0;
 
-        // Add random sleep time between 0.5 and 2.5 seconds
-        sleep(randomInt(500, 2500));
-        var firstQueryId = JSON.stringify(res.query_id);
+      // Call retry in random (between 0.5 and 2.5) seconds
+      setTimeout(function() {
+        retryTweetQuery(tweetInfo, res.query_id);
+      }, randomInt(500, 2500));
 
-        if (counter > MAX_RETRIES) {
-          counter = 1;
-          return;
-        } else {
-          counter++;
-          client.getResponseTweetInfo(firstQueryId).then(function(res) {
-            
-            // Result from second API call
-            var secondCallStatus = JSON.stringify(res.status).replace(/['"]+/g, '');
-
-            if (secondCallStatus.localeCompare('done') === 0) {
-              // As the tweet has been analized then remove the loading spinner
-              var secondRes = JSON.stringify(res);
-
-              var acurracyLabel = JSON.stringify(res.response.rule_engine.final_credibility);
-              console.log("LABEL = " + acurracyLabel);
-
-              // Tweet analized
-              usersCache[tweetInfo.id] = true;
-              tweetInfo.analyzed = true;
-              classifyTweet(tweetInfo, acurracyLabel);
-
-            } else {
-              // Keep showing spinner, close the request and make another request
-              return;
-            }
-          }).catch(err => console.log(err));
+    } else {
+
+      logger.logConsoleDebug(CoInformLogger.logTypes.info, `Done response`, tweetInfo.id);
+
+      // Result from first API call
+      // let firstRes = JSON.stringify(res);
+      let acurracyLabel = JSON.stringify(res.response.rule_engine.final_credibility).replace(/\s+/g,'_');
+      // logger.logConsoleDebug(CoInformLogger.logTypes.info, `LABEL = ${acurracyLabel}`, tweetInfo.id);
+
+      // Tweet analyzed
+      pluginCache[tweetInfo.id] = acurracyLabel;
+      tweetInfo.domObject.coInfoAnalyzed = true;
+      classifyTweet(tweetInfo, acurracyLabel);
+
+    }
+  }).catch(err => {
+    logger.logConsoleDebug(CoInformLogger.logTypes.error, `Request Error: ${err}`, tweetInfo.id);
+    // console.error(err);
+  });
+
+};
+
+const retryTweetQuery = (tweetInfo, queryId) => {
+
+  if (tweetInfo.domObject.coInfoCounter === undefined) {
+    tweetInfo.domObject.coInfoCounter = 0;
+  }
+
+  if (tweetInfo.domObject.coInfoCounter > MAX_RETRIES) {
+
+    logger.logConsoleDebug(CoInformLogger.logTypes.warning, `MAX retries situation (${tweetInfo.domObject.coInfoCounter}). Giving up on tweet..`, tweetInfo.id);
+    return false;
+
+  } else {
+
+    tweetInfo.domObject.coInfoCounter++;
+
+    chrome.runtime.sendMessage({
```

Comment:
In future versions we should make this browser agnostic (i.e. make it work also in Firefox browser)

Reply:
Totally agree. There is the more appropriate similar method called "browser.runtime.sendMessage(...)". Just for note, Firefox is also implementing the chrome object, so it should also work in Firefox.